unstructured/partition/lang.py

```diff
@@ -167,34 +167,68 @@ def prepare_languages_for_tesseract(languages: Optional[List[str]] = ["eng"]):
+    # --- Clean and update defaults
+    if ocr_languages == "auto":
+        raise ValueError(
+            "`ocr_languages` is deprecated but was used to extract text from pdfs and images. "
+            "The 'auto' argument is only for language *detection* when it is assigned "
+            "to `languages` and partitioning documents other than pdfs or images. "
+            "Language detection is not currently supported in pdfs or images."
+        )
 
-    if not isinstance(languages, list):
+    if ocr_languages:
+        ocr_languages = _clean_ocr_languages_arg(ocr_languages)
+        logger.warning(
+            "The ocr_languages kwarg will be deprecated in a future version of unstructured. "
+            "Please use languages instead.",
+        )
+
+    # raise error if languages is not None of a list
+    if not (isinstance(languages, list) or languages is None):
         raise TypeError(
             "The language parameter must be a list of language codes as strings, ex. ['eng']",
         )
 
-    if ocr_languages is not None:
-        if languages != ["eng"]:
-            raise ValueError(
-                "Only one of languages and ocr_languages should be specified. "
-                "languages is preferred. ocr_languages is marked for deprecation.",
-            )
+    # --- If `languages` is a null/default value and `ocr_languages` is defined, use `ocr_languages`
+    if ocr_languages and (
+        languages == ["auto"] or languages == [""] or languages is None or not languages[0]
+    ):
+        languages = ocr_languages.split(TESSERACT_LANGUAGES_SPLITTER)
+        logger.warning(
+            "Only one of languages and ocr_languages should be specified. "
+            "languages is preferred. ocr_languages is marked for deprecation.",
+        )
 
+    # --- Clean `languages`
+    # If "auto" is included in the list of inputs, language detection will be triggered downstream.
+    # The rest of the inputted languages are ignored.
+    if languages:
+        if "auto" not in languages:
+            for i, lang in enumerate(languages):
+                languages[i] = TESSERACT_LANGUAGES_AND_CODES.get(lang.lower(), lang)
+
+            str_languages = _clean_ocr_languages_arg(languages)
+            languages = str_languages.split(TESSERACT_LANGUAGES_SPLITTER)
+        # else, remove the extraneous languages.
+        # NOTE (jennings): "auto" should only be used for partitioners OTHER THAN `_pdf` or `_image`
         else:
-            languages = convert_old_ocr_languages_to_languages(ocr_languages)
-            logger.warning(
-                "The ocr_languages kwarg will be deprecated in a future version of unstructured. "
-                "Please use languages instead.",
-            )
-    return languages
+            # define as 'auto' for language detection when partitioning non-pdfs or -images
+            languages = ["auto"]
+        return languages
+
+    return None
```

Comment:
What do you think about returning a default value (`["eng"]`) instead of `None`, which would avoid setting the default value in `pdf.py` and `image.py`?

Reply:
I don't think we can do that because of the other partitioners that default to "auto". `None` is used as the default because we have to be able to resolve the languages/ocr_languages (or just languages) at the auto `partition` level. Downstream, once the document type is determined and the document is passed to the appropriate partitioner, the default for `languages` can be either "auto" or "eng". "eng" for the other partitioners would skip language detection since it'd be interpreted as a user telling us they know the document is in English.

Reply:
> I don't think we can do that because of the other partitioners that default to "auto". `None` is used as the default because we have to be able to resolve the languages/ocr_languages (or just languages) at the auto `partition` level. Downstream, once the document type is determined and the document is passed to the appropriate partitioner, the default for `languages` can be either "auto" or "eng". "eng" for the other partitioners would skip language detection since it'd be interpreted as a user telling us they know the document is in English.

@Coniferish @cragwolfe @qued
- By this PR, `None` is used as the default at the auto partition level. Downstream, once the document type is determined and the document is passed to the appropriate partitioner, the `None` for `languages` is passed to all partitioners.
- We already have logic to convert `None` for languages to `auto` in `lang.apply_lang_metadata()` - https://github.com/Unstructured-IO/unstructured/blob/main/unstructured/partition/lang.py#L351-L352

Considering the above two, I think it would be better to set the default value for `languages` to `None` rather than `['auto']` for all partitioners in a future refactoring PR. What do you think?